packages/@uppy/companion/src/companion.js

```diff
@@ -212,25 +212,33 @@ const getOptionsMiddleware = (options) => {
   if (options.providerOptions.s3) {
     const S3 = require('aws-sdk/clients/s3')
     const AWS = require('aws-sdk')
-    const config = options.providerOptions.s3
+    const s3ProviderOptions = options.providerOptions.s3
+    const s3Options = Object.assign({
```

Comment:
should we rather introduce a field that covers these options in `providerOptions.s3` itself? So maybe `providerOptions.s3.awsClientOptions` (or something better)?

Reply:
that's a good idea! will update